ctk/common/src/main/kotlin/org/codice/compliance/verification/core/ResponseProtocolVerifier.kt

```diff
@@ -15,155 +15,102 @@ package org.codice.compliance.verification.core
 import org.w3c.dom.Node
 
-class ResponseProtocolVerifier(private val response: Node,
-                               private val id: String,
-                               private val acsUrl: String?) {
+abstract class ResponseProtocolVerifier(open val response: Node,
+                                        open val id: String,
+                                        open val acsUrl: String?) {
     companion object {
-        private const val ID = "ID"
-        private const val VALUE = "Value"
-        private const val STATUS = "Status"
         private const val VERSION = "Version"
-        private const val RESPONSE = "Response"
+        private const val DESTINATION = "Destination"
         private const val STATUS_CODE = "StatusCode"
-        private const val ISSUE_INSTANT = "IssueInstant"
-        private const val SAMLCore_3_2_2 = "SAMLCore.3.2.2"
     }
-    /**
-     * Verify protocols against the Core Spec document
-     * 3.2.2 Complex Type StatusResponseType
-     */
-    fun verify() {
+
+    abstract fun verify()
+
+    /** 3.2.2 Complex Type StatusResponseType */
+    fun verifyCoreResponse() {
         CoreVerifier(response).verify()
         verifyStatusResponseType()
         verifyStatusesType()
-        verifyNameIdMappingResponse()
-
-        if (response.localName == "AuthnRequest") {
-            response.children("Assertion")
-                    .forEach {
-                        if (it.children("AuthnStatement").isEmpty())
-                            throw SAMLComplianceException.create(SAMLCore_3_4,
-                                    message = "AuthnStatement not found.",
-                                    node = response)
-                    }
-        }
+        verify()
     }
 
-    /**
-     * Verify the Status Response Type
-     * 3.2.2 Complex Type StatusResponseType
-     *
-     * All SAML responses are of types that are derived from the StatusResponseType complex type.
-     */
+    /** All SAML responses are of types that are derived from the StatusResponseType complex type.*/
     private fun verifyStatusResponseType() {
-        if (response.attributes.getNamedItem(ID) == null)
-            throw SAMLComplianceException.createWithXmlPropertyReqMessage(SAMLCore_3_2_2,
-                    property = ID,
-                    parent = RESPONSE,
-                    node = response)
-        CommonDataTypeVerifier.verifyIdValues(response.attributes.getNamedItem(ID),
+        CommonDataTypeVerifier.verifyIdValues(response.attributes.getNamedItem("ID"),
                 SAMLCore_3_2_2_a)
 
         // Assuming response is generated in response to a request
         val inResponseTo = response.attributes?.getNamedItem("InResponseTo")?.textContent
-        if (inResponseTo != id)
+        if (inResponseTo != null && inResponseTo != id)
             throw SAMLComplianceException.createWithPropertyMessage(SAMLCore_3_2_2_b,
```

Comment:
I agree with adding a negative test. Are you saying we shouldn't test this then? Cause this will cover the:
```
Otherwise, it MUST be present and its value MUST match the value of the corresponding request's ID attribute.
```